### PR TITLE
Release 1.0.10

### PR DESCRIPTION
**Pull request by ulidtko**
As followup to:
 * #79
 * #82

### CODE REVIEW

File: cabal-doctest.cabal [hunk 1]
```diff
@@ -33,33 +33,30 @@ tested-with:
...
-    -- upper-bounds are major-major only, unless there is evidence for build failures
+    -- NOTE: contrary to PVP, some upper-bounds are intentionally set to major-major.
+    -- This is to increase signal-to-noise ratio of CI failures. "Too tight bounds"
+    -- is an extremely boring (and practically guaranteed, repeatedly) failure mode.
+    -- OTOH, genuine build failures due to breaking changes in dependencies are:
+    --  1) unlikely to occur, as this package is so small, moreso regularly;
+    --  2) best caught in CI pipelines that don't induce alert fatigue.
+    -- In any case, revisions may set tighter bounds afterwards, if exceptional
+    -- circumstances would warrant that.
       base       >=4.3  && <5
```
Comment (andreasabel) [inline cabal-doctest.cabal:55]: Shouldn't you raise to `base >= 4.9` (dropping GHC 7)?

File: cabal-doctest.cabal [hunk 2]
```diff
@@ -33,33 +33,30 @@ tested-with:
...
   build-depends:
-    -- upper-bounds are major-major only, unless there is evidence for build failures
+    -- NOTE: contrary to PVP, some upper-bounds are intentionally set to major-major.
+    -- This is to increase signal-to-noise ratio of CI failures. "Too tight bounds"
+    -- is an extremely boring (and practically guaranteed, repeatedly) failure mode.
+    -- OTOH, genuine build failures due to breaking changes in dependencies are:
+    --  1) unlikely to occur, as this package is so small, moreso regularly;
+    --  2) best caught in CI pipelines that don't induce alert fatigue.
+    -- In any case, revisions may set tighter bounds afterwards, if exceptional
+    -- circumstances would warrant that.
```
Comment (andreasabel) [inline cabal-doctest.cabal:54]: Agreed!

**Review by andreasabel — APPROVED** [review]
LGTM!

NB: In principle, all these changes should be revisionable (staying at 1.0.9).

File: cabal-doctest.cabal (hunk 1 above)
Comment (ulidtko) [reply cabal-doctest.cabal]: Good catch, thanks... fixed.

**Comment by ulidtko** [comment]
Published [v1.0.10](https://hackage.haskell.org/package/cabal-doctest-1.0.10).

@qqwy I don't expect any issues when updating, but please LMK if any difficulty whatsoever arises.

@andreasabel I can agree this technically could've been a revision of 1.0.9 — but as a consumer of the package, I would really expect a genuine version bump :) Rare things are worse than "same but different" versions...

**Comment by andreasabel** [comment]
Yes, as soon as you raise a lower bound (like `base` to 4.9`) there should be a version bump, not a revision.  
In general, only relaxations and corrections of actually wrong bounds may be revisions.  Otherwise correct build plans will be destroyed.

**Comment by ulidtko** [comment]
@andreasabel just curious, do you mean that build plans get persistently saved somehow?.. My understanding up until now was, that build-plans are "ephemeral", produced by cabal solver on-demand, recomputed from package index state + constraints every time anew. Isn't that so?.. Destroyed how?

I'd checked [hackage-trustees guidelines](https://github.com/haskell-infra/hackage-trustees/blob/master/revisions-information.md#what-cant-revisions-change) — and there too, it says anything in the sdist tarball (e.g. README) except some fields of the cabal-file, can't change in revisions. Per these guidelines, commit 009a688d7bffc8df368c51b46ca553cd15e23440 should've been a version bump:

```diff
─────────────────────────────────────────────────────────
modified: src/Distribution/Extra/Doctest.hs
─────────────────────────────────────────────────────────
@ src/Distribution/Extra/Doctest.hs:128 @ import Distribution.Types.LibraryName
        (libraryNameString)
 #endif
 
-#if MIN_VERSION_Cabal(3,6,0)
+#if MIN_VERSION_Cabal(3,5,0)
 import Distribution.Utils.Path
        (getSymbolicPath)
 #endif
@ src/Distribution/Extra/Doctest.hs:334 @ generateBuildModule testSuiteName flags pkg lbi = do
               <- mapM makeAbsolute
                $ compAutogenDir           -- autogenerated files
                : (distPref ++ "/build")   -- preprocessed files (.hsc -> .hs); "build" is hardcoded in Cabal.
-#if MIN_VERSION_Cabal(3,6,0)
+#if MIN_VERSION_Cabal(3,5,0)
                : map getSymbolicPath (hsSourceDirs compBI)
 #else
                : hsSourceDirs compBI
```

because that's a source-code change, not just cabal-file metadata.

In the same trustees repo, there's a [policy.md](https://github.com/haskell-infra/hackage-trustees/blob/master/policy.md#1-metadata-only-changes-tightening-constraints) that apparently encourages retroactive bounds tightening. It also says bounds weakening "should be done carefully"... and doesn't rule it out.

I'd also read this in [cookbook.md](https://github.com/haskell-infra/hackage-trustees/blob/master/cookbook.md):

> Another problem [...] If such install-plans are retroactively prohibited, this results in a bad user-experience, and results in violation of the invariant that a meta-data revision shall not result in existing install-plans becoming illegal.

Are you referring to this issue? When from a user perspective, something used to build, but after a revision somewhere in dependencies, cabal solver would give up?

**Pull request merged**